app/frontend/src/features/messages/surfing/NewHostRequest.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Typography } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Skeleton } from "@material-ui/lab";
+import React, { useEffect } from "react";
+import { useMutation } from "react-query";
+import Alert from "../../../components/Alert";
+import { CreateHostRequestReq } from "../../../pb/requests_pb";
+import { service } from "../../../service";
+import { useUser } from "../../userQueries/useUsers";
+import { Error as GrpcError } from "grpc-web";
+import { useHistory, useParams } from "react-router-dom";
+import { firstName } from "../../../utils/names";
+import { useForm } from "react-hook-form";
+import TextField from "../../../components/TextField";
+import { validateFutureDate } from "../../../utils/validation";
+import Button from "../../../components/Button";
+import { messagesRoute } from "../../../AppRoutes";
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    "& > *": {
+      marginTop: theme.spacing(2),
+    },
+  },
+}));
+
+export default function NewHostRequest() {
+  const classes = useStyles();
+
+  const userId = +useParams<{ userId: string }>().userId;
+
+  const { data: host, isLoading: hostLoading, error: hostError } = useUser(
+    userId
+  );
+  const title = host
+    ? `Request to be hosted by ${firstName(host.name)}`
+    : undefined;
+
+  const { register, handleSubmit, errors: formErrors } = useForm<
+    Required<CreateHostRequestReq.AsObject>
+  >({ defaultValues: { toUserId: userId } });
+
+  useEffect(() => register("toUserId"));
+
+  const history = useHistory();
+
+  const mutation = useMutation<
+    number,
+    GrpcError,
+    Required<CreateHostRequestReq.AsObject>
+  >(
+    (data: Required<CreateHostRequestReq.AsObject>) =>
+      service.requests.createHostRequest(data),
+    {
+      onMutate: console.log,
+      onError: console.log,
```

Comment:
```suggestion
```
Unneeded once you finished developing and checked it works since it's not doing anything extra?